lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -88,6 +94,16 @@ impl Default for wasmer_engine_t {
     }
 }
 
+/// Opaque representing a middleware.
+///
+#[cfg(feature = "middlewares")]
+#[derive(Debug)]
+#[repr(C)]
```

Comment:
We want `wasmer_module_middle_t` to be an opaque type. No need to get a C representation.